lib/simple_form/form_builder.rb

```diff
@@ -674,7 +674,7 @@ def attempt_mapping(mapping, at)
       return if SimpleForm.inputs_discovery == false && at == Object
 
       begin
-        at.const_get(mapping)
+        at.const_get(mapping) if at.constants.include?(mapping.to_sym)
```

Comment:
Are you able to test this with `const_defined?`?

```suggestion
        at.const_get(mapping) if at.const_defined?(mapping)
```

I believe this might achieve similar results without the name error handling, but also without having to do sym conversion & array search.

Reply:
Thanks you for your response. Indeed it works perfectly with with `const_defined?`. It seems that we even gain a few milliseconds
development environment without fix **5105.5ms**
```
Completed 200 OK in 5133ms (Views: 5105.5ms | ActiveRecord: 17.7ms | Allocations: 2132426)
```
development environment first fix *895.0ms*
```
Completed 200 OK in 925ms (Views: 895.0ms | ActiveRecord: 17.1ms | Allocations: 746283)
```
development environment your fix *826.7ms*
```
Completed 200 OK in 853ms (Views: 826.7ms | ActiveRecord: 17.8ms | Allocations: 745690)
```